### PR TITLE
AWS VPC component works with default args.

### PR DESCRIPTION
**Pull request by jkodroff**

### CODE REVIEW

File: awsx/ec2/vpc.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        );
+
+        // We unconditionally create the IGW (even if it's not needed because we
+        // only have isolated subnets) because AWS does not charge for it, and
+        // therefore there's no harm in adding it, whereas conditional resources
+        // must be declared in the constructor, which significantly complicates the
+        // code.
+        const igw = new aws.ec2.InternetGateway(`${name}`, {
+            vpcId: vpc.id,
+        }, { parent: this });
```
Comment (jaxxstorm) [inline awsx/ec2/vpc.ts:80]: the parent of this should be `this.vpc.id` I think

File: awsx/ec2/vpc.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                        },
+                    }, { parent: this });
+                    subnets.push(subnet);
+
+                    const routeTable = new aws.ec2.RouteTable(spec.subnetName, {
+                        vpcId: vpc.id,
+                        tags: {
+                            "Name": spec.subnetName,
+                        },
+                    }, { parent: this });
```
Comment (jaxxstorm) [inline awsx/ec2/vpc.ts:110]: parent to vpc?

File: awsx/ec2/vpc.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                        natGateways.push(natGateway);
+                    }
+
+                    if (spec.type === "Public") {
+                        // Public subnets communicate directly with the internet via the Internet Gateway.
+                        const route = new aws.ec2.Route(spec.subnetName, {
+                            routeTableId: routeTable.id,
+                            gatewayId: igw.id,
+                            destinationCidrBlock: "0.0.0.0/0",
+                        }, { parent: this });
```
Comment (jaxxstorm) [inline awsx/ec2/vpc.ts:144]: parent to route table?

File: awsx/ec2/vpc.ts [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                        // first via the sort above, we know the necessary NAT Gateway already exists.
+                        const natGatewayId = natGatewayStrategy === "Single"
+                            ? natGateways[0].id
+                            : natGateways[i].id;
+
+                        const route = new aws.ec2.Route(spec.subnetName, {
+                            routeTableId: routeTable.id,
+                            natGatewayId,
+                            destinationCidrBlock: "0.0.0.0/0",
+                        }, { parent: this });
```
Comment (jaxxstorm) [inline awsx/ec2/vpc.ts:159]: parent to route table?

File: awsx/ec2/vpc.ts (hunk 1 above)
Comment (jkodroff) [reply awsx/ec2/vpc.ts:80]: Agreed.

Comment (jkodroff) [reply awsx/ec2/vpc.ts:80]: Will be in next push.

**Pull request merged**